File: typescript/packages/bench/src/rules/customerCreditRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class CustomerCreditRule implements RuleContract {
    readonly key = "customer-credit";

    evaluate(data: Dataset): boolean {
        return data.customers.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/customer_credit_rule.rs
//! Rule 8/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct CustomerCreditRule;

impl RuleContract for CustomerCreditRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.customers.is_empty()
    }
}
